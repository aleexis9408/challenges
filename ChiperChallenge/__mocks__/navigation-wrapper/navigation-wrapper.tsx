import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';

jest.useFakeTimers();

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({children, initialParams}: any) => (
  <NavigationContainer>
    <Stack.Navigator
      screenOptions={{
        headerShown: false,
      }}>
      <Stack.Screen
        name="default"
        component={children}
        initialParams={initialParams}
      />
    </Stack.Navigator>
  </NavigationContainer>
);
