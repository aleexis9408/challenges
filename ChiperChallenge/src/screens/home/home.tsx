import React, {useState, useEffect, useContext} from 'react';
import {
  View,
  FlatList,
  RefreshControl,
  ActivityIndicator,
  StyleSheet,
  Dimensions,
} from 'react-native';
import {CardPost} from '../../components/molecules/card-post/card-post';
import {Tabs} from '../../components/molecules/tabs/tabs';
import {RedditServices} from '../../services/Reddit/Reddit.services';
import type {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RootNavigationProps} from '../../navigation/root';
import {IReddit, Ithing} from '../../services/Reddit/Reddit.dto';
import {AppContext} from '../../context/app.provider';
const screen = Dimensions.get('screen');

export const Home = ({}: NativeStackScreenProps<
  RootNavigationProps,
  'home'
>) => {
  const [listData, setListData] = useState<Ithing | undefined>();
  const [refreshing, setRefreshing] = React.useState(false);
  const {select} = useContext(AppContext);
  const TABS = [
    {
      file: 'new',
      title: 'New',
    },
    {
      file: 'top',
      title: 'Top',
    },
    {
      file: 'controversial',
      title: 'Popular',
    },
    {
      file: 'hot',
      title: 'Hot',
    },
  ];

  const getRedditData = async category => {
    try {
      setListData([]);
      const response: IReddit = await RedditServices.getRedditData(category);
      setListData(response.data.children);
    } catch (error) {}
  };

  const onRefresh = async () => {
    setRefreshing(true);
    setListData([]);
    await getRedditData(select);
    setRefreshing(false);
  };

  useEffect(() => {
    getRedditData(select);
  }, [select]);

  return (
    <View style={styles.home}>
      <FlatList
        data={listData}
        keyExtractor={(item, index) => `${index}`}
        refreshControl={
          <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
        }
        ListEmptyComponent={() => (
          <View style={styles.home__spinner}>
            <ActivityIndicator size="large" />
          </View>
        )}
        refreshing={true}
        ListHeaderComponent={() => (
          <React.Fragment>
            <Tabs tabs={TABS} />
          </React.Fragment>
        )}
        renderItem={({item, index}) => (
          <CardPost key={index} content={item.data} />
        )}
      />
    </View>
  );
};

export const styles = StyleSheet.create({
  home: {
    flex: 1,
  },
  home__spinner: {
    flex: 3,
    alignItems: 'center',
    justifyContent: 'center',
    height: screen.height - 200,
  },
});
