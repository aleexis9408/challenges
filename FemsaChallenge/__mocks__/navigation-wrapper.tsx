import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';

jest.useFakeTimers();

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({children, initialParams}: any) => {
  const SomeComponentWrapper = () => children;
  return (
    <NavigationContainer>
      <Stack.Navigator
        screenOptions={{
          headerShown: false,
        }}>
        <Stack.Screen
          name="default"
          component={SomeComponentWrapper}
          initialParams={initialParams}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};
