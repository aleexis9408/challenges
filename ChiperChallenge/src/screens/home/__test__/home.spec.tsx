import React from 'react';
import {render} from '@testing-library/react-native';
import {NavigationWrapper} from '../../../../__mocks__/navigation-wrapper/navigation-wrapper';
import {Home} from '../home';

describe('Render Screen default', () => {
  it('Snapshot', () => {
    const {toJSON} = render(<NavigationWrapper children={Home} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
