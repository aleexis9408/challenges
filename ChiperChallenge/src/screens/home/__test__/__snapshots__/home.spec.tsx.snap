// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Screen default Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            ListEmptyComponent={[Function]}
            ListHeaderComponent={[Function]}
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            refreshControl={
              <RefreshControlMock
                onRefresh={[Function]}
                refreshing={false}
              />
            }
            refreshing={true}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <RCTRefreshControl />
            <View>
              <View
                onLayout={[Function]}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "#cccaca",
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "row",
                      "height": 50,
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "stretch",
                        "backgroundColor": "#37a4be",
                        "fontWeight": "bold",
                        "justifyContent": "center",
                        "opacity": 1,
                        "textAlign": "center",
                        "width": "25%",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 14,
                          },
                          Object {
                            "fontWeight": "bold",
                          },
                        ]
                      }
                    >
                      New
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "stretch",
                        "backgroundColor": "#f2f0f0",
                        "justifyContent": "center",
                        "opacity": 1,
                        "textAlign": "center",
                        "width": "25%",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 14,
                          },
                          false,
                        ]
                      }
                    >
                      Top
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "stretch",
                        "backgroundColor": "#f2f0f0",
                        "justifyContent": "center",
                        "opacity": 1,
                        "textAlign": "center",
                        "width": "25%",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 14,
                          },
                          false,
                        ]
                      }
                    >
                      Popular
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "stretch",
                        "backgroundColor": "#f2f0f0",
                        "justifyContent": "center",
                        "opacity": 1,
                        "textAlign": "center",
                        "width": "25%",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 14,
                          },
                          false,
                        ]
                      }
                    >
                      Hot
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 3,
                    "height": 1134,
                    "justifyContent": "center",
                  }
                }
              >
                <ActivityIndicator
                  size="large"
                />
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
